src/test/java/com/moorhuhnservice/moorhuhnservice/ConfigControllerTest.java

```diff
@@ -0,0 +1,257 @@
+package com.moorhuhnservice.moorhuhnservice;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.moorhuhnservice.moorhuhnservice.data.Configuration;
+import com.moorhuhnservice.moorhuhnservice.data.ConfigurationDTO;
+import com.moorhuhnservice.moorhuhnservice.data.Question;
+import com.moorhuhnservice.moorhuhnservice.data.QuestionDTO;
+import com.moorhuhnservice.moorhuhnservice.data.mapper.ConfigurationMapper;
+import com.moorhuhnservice.moorhuhnservice.data.mapper.QuestionMapper;
+import com.moorhuhnservice.moorhuhnservice.repositories.ConfigurationRepository;
+import com.moorhuhnservice.moorhuhnservice.repositories.QuestionRepository;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@AutoConfigureMockMvc
+@SpringBootTest
+class ConfigControllerTest {
+
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private ConfigurationMapper configurationMapper;
+
+  @Autowired
+  private QuestionMapper questionMapper;
+
+  @Autowired
+  private ConfigurationRepository configurationRepository;
+
+  @Autowired
+  private QuestionRepository questionRepository;
+
+  private final String API_URL = "/api/v1/minigames/moorhuhn/configurations";
+  private ObjectMapper objectMapper;
+  private Configuration createdConfiguration;
+  private ConfigurationDTO createdConfigurationDTO;
+
+  @BeforeEach
+  public void createBasicData() {
+    configurationRepository.deleteAll();
+    final Question questionOne = new Question();
+    questionOne.setText("Are you cool?");
+    questionOne.setRightAnswer("Yes");
+    questionOne.setWrongAnswers(Arrays.asList("No", "Maybe"));
+
+    final Question questionTwo = new Question();
+    questionTwo.setText("Is this game cool?");
+    questionTwo.setRightAnswer("Yes");
+    questionTwo.setWrongAnswers(Arrays.asList("No", "Maybe"));
+
+    final Configuration configuration = new Configuration();
+    configuration.setQuestions(Set.of(questionOne, questionTwo));
+
+    createdConfiguration = configurationRepository.save(configuration);
+    createdConfigurationDTO = configurationMapper.configurationToConfigurationDTO(createdConfiguration);
+
+    objectMapper = new ObjectMapper();
+  }
+
+  @AfterEach
```

Comment:
This will not work because I expect on every database an empty database with my created objects, because I make size asserts like this: `assertSame(2, configurationRepository.findAll().size());` (e.g. after I created a configuration)

Reply:
You already have that functionality in `BeforeEach`, that's the reason why I changed it to ensure an empty database at the end of testing.